app/View/Components/Community/UserCard.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\View\Components\Community;
+
+use App\Community\Enums\Rank;
+use App\Community\Enums\RankType;
+use App\Site\Enums\Permissions;
+use App\Site\Models\User as UserModel;
+use App\Support\Cache\CacheKey;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\View\Component;
+
+class UserCard extends Component
+{
+    private string|array $user;
+
+    public function __construct(string|array $user)
+    {
+        $this->user = $user;
+    }
+
+    public function render(): ?View
+    {
+        $username = $this->getUsername($this->user);
+
+        if (empty($username)) {
+            return null;
+        }
+
+        /* @var array $rawUserData */
+        $rawUserData = [];
+        if (is_array($this->user)) {
+            $rawUserData = $this->user;
+        } else {
+            $rawUserData = $this->getUserData($username);
+        }
+
+        if (!$rawUserData) {
+            return null;
+        }
+
+        $cardViewValues = $this->buildAllCardViewValues($username, $rawUserData);
+
+        return view('components.community.user-card', $cardViewValues);
+    }
+
+    private function getUsername(string|array $user): ?string
+    {
+        return is_string($user) ? $user : ($user['User'] ?? null);
+    }
+
+    private function getUserData(string $username): ?array
+    {
+        return Cache::store('array')->remember(
+            CacheKey::buildUserCardDataCacheKey($username . "1"),
+            Carbon::now()->addMonths(3),
+            function () use ($username): ?array {
+                $foundUser = UserModel::firstWhere('User', $username);
+
+                return $foundUser ? $foundUser->toArray() : null;
+            }
+        );
+    }
+
+    private function buildAllCardViewValues(string $username, array $rawUserData): array
+    {
+        $cardBioData = $this->buildCardBioData($rawUserData);
+        $cardRankData = $this->buildCardRankData($username, $rawUserData['RAPoints'], $rawUserData['RASoftcorePoints'], $rawUserData['Untracked'] ? true : false);
+        $cardRoleData = $this->buildCardRoleData($username, (int) $rawUserData['Permissions']);
```

Comment:
phpunit in CI is failing without this `(int)` cast. However, locally everything passes. I'm not entirely clear why this would be the case. Maybe there is a subtle difference in how something is configured in our GitHub Actions job?

Reply:
I'm not sure what might differ between local and CI, but instead of casting here, you probably want to add `'Permissions' => 'integer'` in the model's `casts` array.

Reply:
Wow, that worked. Good to know. Updated in https://github.com/RetroAchievements/RAWeb/pull/1674/commits/87b59e67ccb06ac73265d9fd3adbc75ac87d2ed4.